airflow/providers/amazon/aws/hooks/logs.py

```diff
@@ -103,7 +104,12 @@ def get_log_events(
                 skip -= event_count
                 events = []
 
-            yield from events
+            if not start_from_head:
+                # if we are not reading from head, it doesn't make sense to return events in "normal" order
+                # while hiding the subsequent calls, bc 1-9 queried by batches of 3 would return 789 456 123
+                yield from reversed(events)
```

Comment:
this is a breaking change, but the method was never used with start_from_head=False so far, and if it was, the results would have been messed up as soon as there was enough logs to make it 2 queries, so I'm calling this a bugfix...

Reply:
What I can propose to make this non-breaking here is to deprecate this parameter which is somewhat broken for the purpose of this function (which is to facilitate making several calls to logs when there are too many for one),

and implement fetching from the bottom using the boto function directly in the ecs hook rather than using this helper. I'd do 1 call only, but the purpose is either to get the very last message, or [to get a couple logs to display in case of an error](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/ecs.py#L668), probably not something that'd require several calls (if we don't get the number of lines required on error, we can just walk with what we have)

Reply:
I did that ☝️ 
tell me if you like it better.